DParser2/Resolver/DTypeToCodeVisitor.cs

```diff
@@ -98,7 +98,27 @@ public void VisitDelegateType(DelegateType t)
 		void VisitDSymbol(DSymbol t)
 		{
 			var def = t.Definition;
-			sb.Append(def != null ? def.ToString(false, true) : "<Node object no longer exists>");
+			if (def == null)
+				sb.Append("<Node object no longer exists>");
+			else
+			{
+				sb.Append(def.Name);
+				if (def.TemplateParameters != null)
+				{
+					// TODO: to match dmd, do not emit "()" if argument is
+					//  string, basic type, single (unresolved) identifier,
+					//  number, "null" or "this"

```

Comment:
Gibt es dafür nicht auch schon zusammengefasste Repräsentationen, die man ggf. über ein type-switch testen kann? 
* ISymbolValue
* PrimitiveType
* das mit dem `single (unresolved) identifier` habe ich nicht so ganz durchschaut.
Aber so könnte man das doch lösen, oder?

Reply:
Der entsprechende dmd code ist hier: https://github.com/dlang/dmd/blob/master/src/dmd/hdrgen.d#L1635
`Tident` sollte es nach der semantischen Analyse aber nicht mehr geben, und struct-Namen werden dann doch in Klammern angezeigt. Optimal und eindeutig ist das ja nicht so richtig.